### ISSUE TITLE
Add createFactoryWithConstrain function utility

### ISSUE THREAD

**Issue by Beraliv** [ts-essentials/ts-essentials#317]
Add new function utility which is useful in tests:

```ts
const createFactoryWithConstrain =
    <Constrain>() =>
    <U extends Constrain>(value: U): U =>
        value;
```

Playground with use cases – [tsplay.dev/Ndo6Yw](https://t.co/JA09vr47Lf)

**Comment by Beraliv** [comment]
Merged it to master branch

I will release it soon in https://github.com/ts-essentials/ts-essentials/pull/326

**Comment by Beraliv** [comment]
Released in 9.3.0 ✅
